enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,461 @@
+4. Provide a way for index authors to reason about their indexes via a visual representation of the index, without having to curl a grpc api that in turn queries a sqllite database underneath, etc.
+5. Provide a way for tools like `opm`, `operator-registry` etc to consume the representation of an index instead of needing a sqlite database as an input whenever they need information about the index to perform a task. 
+6. Allow for reproducibility of an index using just the yaml/json representation of the index. 
+
+### Non-goals 
+
+1. Enumerate/implement tooling to allow different operations to be performed on the package representations. Numerous tooling(eg those that allow operation on files) that already exists can be leveraged to perform various operations on package representations.    
+
+### User Stories 
+
+#### Story 1
+
+As a user with pull permission from the namespace an index is hosted in/as a component of operator-lifecycle-manager, I can query an index for a json/yaml representation of the packages of an index.
+The representation of individual packages will allow individual package owners to reason about/make changes to their individual packages in isolation.  
+
+`opm index inspect --index=docker.io/my-namespace/community-operators --output=json`
+
+creates a new folder `community-operators` and downloads the package representations for the packages that are in the index
+
+```bash
+$ cd community-operators
+$ tree
+.
+├── etcd.json
+├── amqstreams.json
+
+$ cat etcd.json
+"etcd": {
+    name:           "etcd"
+    defaultChannel: "alpha"
+    icon:           "asdfasdfoipuasdfvasdfqer"
+    channels: [{
+        name: "alpha"
+        bundles: [{
+            path:        "quay.io/random/etcdv0.0.1"
+            version:     "v0.0.1"
+            upgradeMode: "semver"
+            ownedAPIs: [{
+                group:   "etcd.database.coreos.com"
+                version: "v1alpha1"
+                kind:    "etcd"
+            }]
+            requiredAPIs: []
+        }, {
+            path:        "quay.io/somethingelse/etcdv0.0.2"
+            version:     "v0.0.2"
+            upgradeMode: "semver"
+            ownedAPIs: [{
+                group:   "etcd.database.coreos.com"
+                version: "v1alpha1"
+                kind:    "etcd"
+            }]
+            requiredAPIs: []
+        }]
+    }]
+}
+
+$ cat amqstream.json
+amqstreams: {
+	name:           "amq-streams"
+	defaultChannel: "alpha"
+	icon:           "asdfasdfoipuasdfvasdfqer"
+	channels: [{
+		name:        "alpha"
+		upgradeMode: "semver"
+		bundles: [{
+			path:    "quay.io/mynamespace/amq-streamsv0.0.1"
+			version: "v0.0.1"
+			ownedAPIs: [{
+				group:   "amq-streams.org.com"
+				version: "v1alpha1"
+				kind:    "amq-streams"
+			}]
+			requiredAPIs: [{
+				group:   "etcd.database.coreos.com"
+				version: "v1alpha1"
+				kind:    "etcd"
+			}]
+		}]
```

Comment:
this api structure means that for a bundle to belong to multiple channels, the entire bundle stanza must be repeated.  That seems unnecessarily redundant.